sktime/datasets/tests/test_data_io.py

```diff
@@ -44,24 +43,23 @@
 
 @pytest.mark.parametrize("dataset_name", ["UnitTest", "BasicMotions"])
 @pytest.mark.parametrize("return_type", ["nested_univ", "numpy3d"])
-def test_write_panel_to_tsfile_equal_length(dataset_name, return_type):
+def test_write_panel_to_tsfile_equal_length(dataset_name, return_type, tmpdir):
```

Comment:
@fkiraly The changes in this file are unrelated to my PR, but I ran into problems with a race condition testing my changes.

Reply:
this is sth I would separate out, actually. Should be a few clicks only.

Would be appreciated if you could describe the race condition somewhere, e.g., in an issue?

Reply:
you can still have it in this PR, and at the top you say, "depends on PR such-and-such". I.e., if the other PR is merged more quickly, the diff in this one gets smaller.

Reply:
Moved to #4192